sdk/python/kfp/v2/components/experimental/pipeline_channel.py

```diff
@@ -0,0 +1,321 @@
+        constructs it directly (for providing an immediate value), or it is a
+        pipeline function argument.
+      pattern: The serialized string regex pattern this pipeline channel created
+        from.
+      value: The actual value of the pipeline channel. If provided, the
+        pipeline channel is "resolved" immediately.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        channel_type: Union[str, Dict],
+        op_name: Optional[str] = None,
+        value: Optional[type_utils.PARAMETER_TYPES] = None,
+    ):
+        """Inits a PipelineArtifactChannel instance.
+
+        Args:
+          name: The name of the pipeline channel.
+          channel_type: The type of the pipeline channel.
+          op_name: Optional; The name of the operation that produces the
+            pipeline channel.
+          value: Optional; The actual value of the pipeline channel.
+
+        Raises:
+          ValueError: If name or op_name contains invalid characters.
+          ValueError: If both op_name and value are set.
+          TypeError: If the channel type is not a parameter type.
+        """
+        if op_name and value:
+            raise ValueError('op_name and value cannot be both set.')
+
+        if not type_utils.is_parameter_type(channel_type):
+            raise TypeError(f'{channel_type} is not a parameter type.')
+
+        self.value = value
+
+        super(PipelineParameterChannel, self).__init__(
+            name=name,
+            channel_type=channel_type,
+            op_name=op_name,
+        )
+
+
+class PipelineArtifactChannel(PipelineChannel):
+    """Represents a pipeline artifact channel.
+
+    Attributes:
+      name: The name of the pipeline channel.
+      channel_type: The type of the pipeline channel.
+      op_name: The name of the operation that produces the pipeline channel.
+        A pipeline artifact channel is always produced by some op.
+      pattern: The serialized string regex pattern this pipeline channel created
+        from.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        channel_type: Union[str, Dict],
+        op_name: str,
+    ):
+        """Inits a PipelineArtifactChannel instance.
+
+        Args:
+          name: The name of the pipeline channel.
+          channel_type: The type of the pipeline channel.
+          op_name: The name of the operation that produces the pipeline channel.
+
+        Raises:
+          ValueError: If name or op_name contains invalid characters.
+          TypeError: If the channel type is not an artifact type.
+        """
+        if type_utils.is_parameter_type(channel_type):
+            raise TypeError(f'{channel_type} is not an artifact type.')
+
+        super(PipelineArtifactChannel, self).__init__(
+            name=name,
+            channel_type=channel_type,
+            op_name=op_name,
+        )
+
+
+def extract_pipeline_channels(payload: str) -> List['PipelineChannel']:
```

Comment:
Still need the forward reference for `PipelineChannel` here?

Reply:
Up to you, but do you want to name this `extract_pipeline_channels_from_string` to make it symmetric with the method name below?